apps/parsley/src/components/Search/index.tsx

```diff
@@ -49,15 +62,32 @@ const Search: React.FC = () => {
       case SearchBarActions.Filter:
         if (!filters.some((f) => f.expression === value)) {
           setSearch("");
-          setFilters([
-            ...filters,
-            {
-              caseSensitive: CaseSensitivity.Insensitive,
-              expression: value,
-              matchType: MatchType.Exact,
-              visible: true,
-            },
-          ]);
+
+          if (highlightFilters) {
+            setSearchParams({
```

Comment:
I'm not sure if this is possible :(. Since these query param updates aren't like useState the query updates aren't batched and thus can't be made individually.